Model/Usage/CatagoriesOptionsProvider.php

```diff
@@ -3,6 +3,9 @@
 
 class CatagoriesOptionsProvider implements \Magento\Framework\Data\OptionSourceInterface
 {
+
+    const CUSTOM_LICENSE = 'Custom License';
```

Comment:
Requiring a category named `Custom License` is not the approach we want to use. The Custom License category should not be a category that actually exists in the DB but is dynamically added to the category list on the front end when custom license are present for the customer. 

There are several downsides to using this approach. Mostly translation of the category name will be difficult doing it this way.

Reply:
we can create a backend configuration where admin can select which category is used for custom license?

Reply:
There should be no need to create a category for custom license. It'll just be an extra option in the categories drop-down but not a real category in the database.

Reply:
@superdav42 the database structure restricts us to push category in category table as the category id is foreign key for usage(s). First thing, the constant with value **Custom License** is only for coding purpose and is not being used to view in category dropdown. It's being fetched from the collection you have in database.

In order to manipulate this constant value, we have the following suggestions, you can choose any of them:

1. Create a configuration at backend in order to let customer choose which category belongs to Custom License.
2. Change the database structure of the custom table and update the behaviour of the current module to allow nullable foreign keys in usage table. We can refer the usages' categories to be a static foreign key value or null.
3. We can create **Custom License** category via UpgradeData file once this module is installed/upgraded but we can restrict the visibility of this **Custom License** category in category grid (it can be hidden or can be non-editable).

Let us know which option you want to go with, I'd say go with #1 as that makes more sense to me.

About the translation part, it can be automatically translated. The **Custom License**  is concerned only for code side, so nothing to worry about there.

Reply:
1. Should be fine for now. I was thinking 2 or maybe even allowing any category to have a combination of usages customer and not customer specific but that is probably much more work. So let's just do the easiest for now.